scarb/src/resolver/mod.rs

```diff
@@ -38,163 +34,15 @@ pub async fn resolve(
-                    DepKind::Normal => None,
-                    DepKind::Target(target_kind) => Some(target_kind),
-                };
-                let dep = dep_summary.package_id;
-
-                if let Some(existing) = packages.get(dep.name.as_ref()) {
-                    if existing.source_id != dep.source_id {
-                        bail!(
-                            indoc! {"
-                            found dependencies on the same package `{}` coming from incompatible \
-                            sources:
-                            source 1: {}
-                            source 2: {}
-                            "},
-                            dep.name,
-                            existing.source_id,
-                            dep.source_id
-                        );
-                    }
-                }
-
-                let weight = graph
-                    .edge_weight(package_id, dep)
-                    .cloned()
-                    .unwrap_or_default();
-                let weight = weight.extend(dep_target_kind);
-                graph.add_edge(package_id, dep, weight);
-                summaries.insert(dep, dep_summary.clone());
-
-                if packages.contains_key(dep.name.as_ref()) {
-                    continue;
-                }
-
-                packages.insert(dep.name.clone(), dep);
-                next_queue.push(dep);
-            }
-        }
-
-        queue = next_queue;
-    }
-
-    // Detect incompatibilities and bail in case ones are found.
-    let mut incompatibilities = Vec::new();
-    for from_package in graph.nodes() {
-        let dep_filter = DependencyFilter::propagation(main_packages.contains(&from_package));
-        for manifest_dependency in summaries[&from_package].filtered_full_dependencies(dep_filter) {
-            let to_package = packages[&manifest_dependency.name];
-            if !manifest_dependency.matches_package_id(to_package) {
-                let message = format!(
-                    "- {from_package} cannot use {to_package}, because {} requires {} {}",
-                    from_package.name, to_package.name, manifest_dependency.version_req
-                );
-                incompatibilities.push(message);
-            }
-        }
-    }
-
-    if !incompatibilities.is_empty() {
-        incompatibilities.sort();
-        let incompatibilities = incompatibilities.join("\n");
-        bail!(formatdoc! {"
-            Version solving failed:
-            {incompatibilities}
-
-            Scarb does not have real version solving algorithm yet.
-            Perhaps in the future this conflict could be resolved, but currently,
-            please upgrade your dependencies to use latest versions of their dependencies.
-        "});
+    let algo_primitive = env::var("SCARB_UNSTABLE_PRIMITIVE_RESOLVER")
+        .ok()
+        .map(|var| var.as_str() == "true")
+        .unwrap_or(true);
```

Comment:
I'd rather model config as "enable" rather than "disable"

```suggestion
    let algo_primitive = env::var("SCARB_PUBGRUB")
        .ok()
        .map(|var| {
            let s = var.as_str();
            s == "true" || s == "1"
        })
        .unwrap_or(false);
```